src/window/city.c

```diff
@@ -99,22 +108,127 @@ static void draw_foreground_military(void)
     draw_paused_and_time_left();
 }
 
-static void handle_mouse(const mouse *m)
+static void exit_military_command(void)
 {
+    if (window_is(WINDOW_CITY_MILITARY)) {
+        window_city_show();
+    }
+}
+
+static void show_overlay(int overlay)
+{
+    exit_military_command();
+    if (game_state_overlay() == overlay) {
+        game_state_set_overlay(OVERLAY_NONE);
+    } else {
+        game_state_set_overlay(overlay);
+    }
+    city_with_overlay_update();
+    window_invalidate();
+}
+
+static void cycle_legion(void)
+{
+    static int current_legion_id = 1;
+    if (window_is(WINDOW_CITY)) {
+        int legion_id = current_legion_id;
+        current_legion_id = 0;
+        for (int i = 1; i <= MAX_LEGIONS; i++) {
```

Comment:
I know this bug isn't directly caused by your changes, but I think it should be `MAX_FORMATIONS` here, not `MAX_LEGIONS`.

Reply:
True. I'll fix that directly on master though.